src/engine/Server.cpp

```diff
@@ -811,11 +811,16 @@ boost::asio::awaitable<void> Server::processQuery(
   //  optional<errorMsg> and optional<metadata> and does this logic
   if (exceptionErrorMsg) {
     LOG(ERROR) << exceptionErrorMsg.value() << std::endl;
-    auto errorResponseJson = composeErrorResponseJson(
-        query, exceptionErrorMsg.value(), requestTimer, metadata);
     if (metadata.has_value()) {
-      LOG(ERROR) << metadata.value().coloredError() << std::endl;
+      try {
+        LOG(ERROR) << metadata.value().coloredError() << std::endl;
+      } catch (const std::exception&) {
+        LOG(ERROR) << "Failed to highlight error in query " << std::endl;
+        LOG(ERROR) << metadata.value().query_ << std::endl;
```

Comment:
I didn't finish my sentence earlier:
You have moved this code to the beginning. So now you can also add the `what()` from the exception together with an "Highlighting an error for the command line log failed " to the `exceptionErrorMsg`, s.t. we also can detect this case from the UI if it still happens.